spec/src/antlr/JDQL.g4

```diff
@@ -0,0 +1,93 @@
+grammar JDQL;
+
+statement : select_statement | update_statement | delete_statement;
+
+select_statement : select_clause? from_clause? where_clause? orderby_clause?;
+update_statement : 'UPDATE' entity_name set_clause where_clause?;
+delete_statement : 'DELETE' from_clause where_clause?;
+
+from_clause : 'FROM' entity_name;
+
+where_clause : 'WHERE' conditional_expression;
+
+set_clause : 'SET' update_item (',' update_item)*;
+update_item : state_field_path_expression '=' (scalar_expression | 'NULL');
+
+select_clause : 'SELECT' select_list;
+select_list
+    : state_field_path_expression
+    | aggregate_expression
+    ;
+aggregate_expression : 'COUNT' '(' '*' ')';
```

Comment:
`COUNT(*)` doesn't appear to be valid JPQL per the BNF that is in Jakarta Persistence 3.2 M2.
https://jakarta.ee/specifications/persistence/3.2/jakarta-persistence-spec-3.2-m2#bnf
Hopefully this is one of the things being added as part of making identification variables optional.

Reply:
Hrm, yeah, OK, in JPQL it would be `count(this)`. You want me to change it to that?

Reply:
Yes, we need to ensure JDQL is a subset JPQL

Reply:
Done.